pkg/clusteragent/custommetrics/provider.go

```diff
@@ -71,6 +74,16 @@ func (p *datadogProvider) ListAllExternalMetrics() []provider.ExternalMetricInfo
 	var externalMetricsInfoList []provider.ExternalMetricInfo
 	var externalMetricsList []externalMetric
 
+	copyAge := metav1.Now().Unix() - p.timestamp
+	if copyAge < p.maxAge { //&& p.externalMetrics != nil{
+		log.Tracef("Local copy is recent enough, not querying the GlobalStore. Remaining %d/%d seconds before next sync", copyAge, p.maxAge)
+		for _, in := range p.externalMetrics {
+			externalMetricsInfoList = append(externalMetricsInfoList, in.info)
+		}
+		return externalMetricsInfoList
+	}
+
+	log.Debugf("Local copy of external metrics from the global store is outdated by %d seconds, resyncing now", copyAge-p.maxAge)
```

Comment:
I think it can be consistent with the log above and be at the `trace` level as well

Reply:
Either way is fine imho, we can justify the `Debug` because this one triggers a refresh when the other one does not, and it seems the other one can get spammy when this one will not.

Reply:
I think we should leave it as Debug, from my experience you know that if you see this log important things are happening. Whereas the one above just indicates a nominal behavior